aten/src/ATen/native/Normalization.cpp

```diff
@@ -44,29 +43,23 @@ Tensor batch_norm(
   }
 
   bool use_cudnn = false;
-#if AT_CUDNN_ENABLED()
   use_cudnn = (input.type().is_cuda()
                && (input.type().scalarType() != at::kHalf
                  || weight.type().scalarType() == at::kFloat)
                && weight.defined() && bias.defined()
                && ((running_mean.defined() && running_var.defined())
                  || (!running_mean.defined() && !running_var.defined() && training))
                && input.size(0) <= 131070
-               && cudnn_enabled && CUDNN_VERSION >= 5110L);
-#else
-  // avoid unused parameter warning
-  (void)use_cudnn;
-  (void)cudnn_enabled;
-#endif
-
-#if AT_CUDNN_ENABLED()
-  if (use_cudnn && eps >= CUDNN_BN_MIN_EPSILON) {
+               && detail::getCUDAHooks().compiledWithCuDNN()
```

Comment:
Would be nice if we avoided calling `getCUDAHooks` too many times. After all every call is an atomic load or some kind of atomic test and set for `call_once`

Reply:
On a TSO memory model (e.g., X86) an atomic load is just a plain old MOV...